sys/include/net/sock/async.h

```diff
@@ -33,7 +33,22 @@ extern "C" {
 #if defined(MODULE_SOCK_DTLS) || defined(DOXYGEN)
 
 /**
- * @brief   Gets the asynchronous event context from sock object
+ * @brief Gets the asynchronous event session from sock object
+ *
+ * @pre `(sock != NULL) && (session != NULL)`
+ *
+ * @note Only available with @ref SOCK_HAS_ASYNC defined.
+ *       Should only be called within a DTLS event and only for the event
+ *       types @ref SOCK_ASYNC_CONN_RDY and @ref SOCK_ASYNC_CONN_FIN.
```

Comment:
Mhhh maybe add a return value, that signals to the caller when the the session is available or not? I know this makes the implementation more complicated than it is now, but `sock` is an end-user API that is supposed to be easy to use. Could be boolean or an `errno`-based status code

Reply:
Totally forgot, sorry. Squashed the undo directly with the related commit. 

Regarding the return value. It would be nice to have it...but how?  The only idea I have for this currently is to also add (e.g.) a boolean to the socket object and set this boolean when a event occurs. So setting it for the CONN_RDY, CONN_FIN to available and for every other event to not available. Would that be sufficient or do you imagine it differently?

Reply:
In the specific case of `tinydtls` you could use [the `size` member of the `session_t` struct](https://github.com/eclipse/tinydtls/blob/706888256c3e03d9fcf1ec37bb1dd6499213be3c/session.h#L40-L49) as a signifier if the session is set.

```C
typedef struct {
  socklen_t size;		/**< size of addr */
  union {
    struct sockaddr     sa;
    struct sockaddr_storage st;
    struct sockaddr_in  sin;
    struct sockaddr_in6 sin6;
  } addr;
  uint8_t ifindex;
} session_t;
```

The object needs to zero'd of course. However, I would do this on read and socket init. This way, if you e.g. receive a CONN_RDY and a MSG_RECV in the same callback you don't accidentally remove the session before it can be read.

Reply:
Done :) Thank you! I've adjusted the test procedure to test, that the session is not available for other event types.